Add TravisCI support.

miguelandres opened the pull request:
-Add readme codecov & status badges.
-De-linting so it doesn't fail on its first run.

Commit:
Add Travis configuration

Commit:
Fix travis lint

Commit:
Set travis to build, test. Add badges

Commit:
De-lint Remixer before travis integration

Commit:
Fix badges, they were pointing to wrong repo

pingpongboss commented on .travis.yml:
```diff
@@ -0,0 +1,28 @@
+    - android-24
+    - build-tools-24.0.1
+    - extra-android-support
+    - extra-android-m2repository
```
can you verify that you need these 2 `extra-android-` components? See if travsici builds without them.

miguelandres replied:
Android support is definitely necessary.

pingpongboss commented on .travis.yml:
```diff
@@ -0,0 +1,28 @@
+  - echo yes | android update sdk --filter extra-android-m2repository --no-ui --force > /dev/null
+
+script:
+  - ./gradlew build test
```
I suggest using `check` which includes `build` and `test` but also includes plugin outputs like jacoco (for code coverage).

miguelandres replied:
done

pingpongboss commented on README.md:
```diff
@@ -1,3 +1,4 @@
+[![Build Status](https://travis-ci.org/material-foundation/material-remixer-android.svg?branch=develop)](https://travis-ci.org/material-foundation/material-remixer-android) [![codecov](https://codecov.io/gh/material-foundation/material-remixer-android/branch/develop/graph/badge.svg)](https://codecov.io/gh/material-foundation/material-remixer-android)
```
we have codecov on a separate line (in source) which helps readability

miguelandres replied:
done

Commit:
Remove before_install and extras

Commit:
Use gradlew check, remove proguarding, readme readability.

-Remove proguarding because it takes a really long time.
-use gradlew check because it includes everything we need
-add a \n in the readme for readability

Commit:
Re-add android-support, it's necessary

Commit:
Re add before_install

Commit:
Update gradle dependency, add lint ignore, tweak travis file

Commit:
mark remixer_example abortOnError false for lint

I can’t debug this lint issue in travis because I can’t reproduce it on
my machine, I imagine there is a different default configuration of
tools here and there.

Also lint on the example app is not terribly useful.

Commit:
Set up text output for lint

Commit:
Try to hack travis file to update the android build tools.

Commit:
Add comments to explain why the weird configuration in the travis file

miguelandres commented:
Finally figured it out! Travis installs old tools by default, and you have to work around it by installing them twice.

https://github.com/travis-ci/travis-ci/issues/6040#issuecomment-219367943

I'll be squashing this PR, since the history is basically useless, a bunch of attempts to diagnose the issue.

pingpongboss reviewed (commented):
Hmm think I should do this for MM?

miguelandres commented:
Probably...it may be that your target api is 23 and it hasn't affected you
but it would if you move to 24/25

Den tors 10 nov. 2016 12:56Mark Wei notifications@github.com skrev:

> _@pingpongboss_ commented on this pull request.
> 
> Hmm think I should do this for MM?
> 
> —
> You are receiving this because you modified the open/close state.
> 
> Reply to this email directly, view it on GitHub
> https://github.com/material-foundation/material-remixer-android/pull/74#pullrequestreview-8063069,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AAJxqD5TIcIIuBHbAqnXjCJrTHIK_UVeks5q81rogaJpZM4KqEBG
> .
